File: python/test_sdk.py
#!/usr/bin/env python3
"""
Test du SDK SahelPay - Python

Ce script teste les nouvelles fonctionnalités:
- Plans
- Subscriptions
- Customers
"""

from sahelpay import Client
import sys
import os
sys.path.insert(0, '.')


SECRET_KEY = os.getenv('SAHELPAY_SECRET_KEY')


def test_sdk():
    print('🚀 Test du SDK SahelPay Python\n')
    print('=' * 50)

    if not SECRET_KEY:
        print('❌ SAHELPAY_SECRET_KEY is required')
        return

    client = Client(secret_key=SECRET_KEY, environment='production')

    try:
        # ==================== TEST PLANS ====================
        print('\n📋 TEST: Plans API')
        print('-' * 30)

        # Lister les plans existants
        print('→ Listing des plans...')
        plans = client.plans.list()
        print(f'✅ {len(plans)} plan(s) trouvé(s)')
        for p in plans:
            print(f"   - {p['name']}: {p['amount']} XOF/{p['interval']}")

        # Créer un nouveau plan
        print('\n→ Création d\'un plan de test...')
        import time
        new_plan = client.plans.create(
            name=f'Python SDK Test {int(time.time())}',
            amount=3000,
            interval='MONTHLY',
        )
        print(f"✅ Plan créé: {new_plan['id']}")
        print(f"   Nom: {new_plan['name']}")
        print(
            f"   Montant: {new_plan['amount']} {new_plan.get('currency', 'XOF')}")

        # ==================== TEST SUBSCRIPTIONS ====================
        print('\n📅 TEST: Subscriptions API')
        print('-' * 30)

        # Lister les abonnements
        print('→ Listing des abonnements...')
        subs_result = client.subscriptions.list()
        subs = subs_result.get('subscriptions', [])
        print(f'✅ {len(subs)} abonnement(s) trouvé(s)')

        # Créer un abonnement de test
        print('\n→ Création d\'un abonnement de test...')
        new_sub = client.subscriptions.create(
            plan_id=new_plan['id'],
            customer_phone='22370000003',
        )
        print(f"✅ Abonnement créé: {new_sub['id']}")
        print(f"   Plan: {new_sub.get('plan_name', new_plan['name'])}")
        print(f"   Client: {new_sub['customer_phone']}")
        print(f"   Statut: {new_sub['status']}")
        print(f"   Prochaine facturation: {new_sub['next_billing_date']}")

        # ==================== TEST CUSTOMERS ====================
        print('\n👥 TEST: Customers API')
        print('-' * 30)

        # Lister les clients
        print('→ Listing des clients...')
        customers_result = client.customers.list()
        customers = customers_result.get('customers', [])
        print(f'✅ {len(customers)} client(s) trouvé(s)')

        # ==================== TEST PAYMENT LINKS ====================
        print('\n🔗 TEST: Payment Links API')
        print('-' * 30)

        print('→ Listing des liens de paiement...')
        links = client.payment_links.list()
        print(f'✅ {len(links)} lien(s) de paiement trouvé(s)')

        # ==================== TEST PAYMENTS ====================
        print('\n💳 TEST: Payments API')
        print('-' * 30)

        print('→ Listing des paiements...')
        payments_result = client.payments.list(limit=5)
        payments = payments_result.get('payments', [])
        print(f'✅ {len(payments)} paiement(s) récent(s)')

        # ==================== RÉSUMÉ ====================
        print('\n' + '=' * 50)
        print('✅ TOUS LES TESTS PASSÉS AVEC SUCCÈS!')
        print('=' * 50)

    except Exception as e:
        print(f'\n❌ ERREUR: {e}')
        if hasattr(e, 'code'):
            print(f'   Code: {e.code}')
        if hasattr(e, 'status_code'):
            print(f'   Status: {e.status_code}')


if __name__ == '__main__':
    test_sdk()

File: javascript/test-sdk.ts
/**
 * Test du SDK SahelPay - JavaScript/TypeScript
 * 
 * Ce script teste les nouvelles fonctionnalités:
 * - Plans
 * - Subscriptions
 * - Customers
 */

import SahelPay from './src/index';

const SECRET_KEY = process.env.SAHELPAY_SECRET_KEY;

async function testSDK() {
  console.log('🚀 Test du SDK SahelPay JavaScript\n');
  console.log('='.repeat(50));

  if (!SECRET_KEY) {
    console.error('❌ SAHELPAY_SECRET_KEY is required');
    process.exitCode = 1;
    return;
  }

  const sahelpay = new SahelPay({
    secretKey: SECRET_KEY,
    environment: 'production',
  });

  try {
    // ==================== TEST PLANS ====================
    console.log('\n📋 TEST: Plans API');
    console.log('-'.repeat(30));

    // Lister les plans existants
    console.log('→ Listing des plans...');
    const plans = await sahelpay.plans.list();
    console.log(`✅ ${plans.length} plan(s) trouvé(s)`);
    plans.forEach(p => console.log(`   - ${p.name}: ${p.amount} XOF/${p.interval}`));

    // Créer un nouveau plan
    console.log('\n→ Création d\'un plan de test...');
    const newPlan = await sahelpay.plans.create({
      name: `SDK Test Plan ${Date.now()}`,
      amount: 2500,
      interval: 'MONTHLY',
    });
    console.log(`✅ Plan créé: ${newPlan.id}`);
    console.log(`   Nom: ${newPlan.name}`);
    console.log(`   Montant: ${newPlan.amount} ${newPlan.currency}`);

    // ==================== TEST SUBSCRIPTIONS ====================
    console.log('\n📅 TEST: Subscriptions API');
    console.log('-'.repeat(30));

    // Lister les abonnements
    console.log('→ Listing des abonnements...');
    const subsResult = await sahelpay.subscriptions.list();
    console.log(`✅ ${subsResult.subscriptions?.length || 0} abonnement(s) trouvé(s)`);

    // Créer un abonnement de test
    console.log('\n→ Création d\'un abonnement de test...');
    const newSub = await sahelpay.subscriptions.create({
      plan_id: newPlan.id,
      customer_phone: '22370000002',
    });
    console.log(`✅ Abonnement créé: ${newSub.id}`);
    console.log(`   Plan: ${newSub.plan_name}`);
    console.log(`   Client: ${newSub.customer_phone}`);
    console.log(`   Statut: ${newSub.status}`);
    console.log(`   Prochaine facturation: ${newSub.next_billing_date}`);

    // ==================== TEST CUSTOMERS ====================
    console.log('\n👥 TEST: Customers API');
    console.log('-'.repeat(30));

    // Lister les clients
    console.log('→ Listing des clients...');
    const customersResult = await sahelpay.customers.list();
    console.log(`✅ ${customersResult.customers?.length || 0} client(s) trouvé(s)`);

    // ==================== TEST PAYMENT LINKS ====================
    console.log('\n🔗 TEST: Payment Links API');
    console.log('-'.repeat(30));

    console.log('→ Listing des liens de paiement...');
    const links = await sahelpay.paymentLinks.list();
    console.log(`✅ ${links.length} lien(s) de paiement trouvé(s)`);

    // ==================== TEST PAYMENTS ====================
    console.log('\n💳 TEST: Payments API');
    console.log('-'.repeat(30));

    console.log('→ Listing des paiements...');
    const paymentsResult = await sahelpay.payments.list({ limit: 5 });
    console.log(`✅ ${paymentsResult.payments.length} paiement(s) récent(s)`);

    // ==================== RÉSUMÉ ====================
    console.log('\n' + '='.repeat(50));
    console.log('✅ TOUS LES TESTS PASSÉS AVEC SUCCÈS!');
    console.log('='.repeat(50));

  } catch (error: any) {
    console.error('\n❌ ERREUR:', error.message);
    if (error.code) console.error('   Code:', error.code);
    if (error.statusCode) console.error('   Status:', error.statusCode);
  }
}

testSDK();
